testing/tests/DevExpress.ui.widgets/listParts/commonTests.js

```diff
@@ -1788,6 +1789,150 @@ QUnit.module('selection', moduleSetup, () => {
     });
 });
 
+QUnit.module('selectByClick option', {
+    beforeEach: function() {
+        this.onSelectionChangedSpy = sinon.spy();
+        this.items = [1, 2];
+        this.createList = (selectionMode, selectByClick) => {
+            return $('#list').dxList({
+                items: this.items,
+                showSelectionControls: true,
+                selectByClick,
+                selectionMode,
+                onSelectionChanged: this.onSelectionChangedSpy,
+            });
+        };
+    } }, ()=> {
+
+    QUnit.test('onSelectionChanged event should call on item click if selectionMode=single, selectByClick=true', function(assert) {
+        const $list = this.createList('single', true);
+        const $items = $list.find(`.${LIST_ITEM_CLASS}`);
+
+        $items.eq(0).trigger('dxclick');
+        assert.strictEqual($items.eq(0).hasClass(LIST_ITEM_SELECTED_CLASS), true, 'item has selected class');
+        assert.strictEqual($items.eq(1).hasClass(LIST_ITEM_SELECTED_CLASS), false, 'item doesn`t have selected class');
+
+        $items.eq(1).trigger('dxclick');
+        assert.strictEqual($items.eq(0).hasClass(LIST_ITEM_SELECTED_CLASS), false, 'item doesn`t have selected class');
+        assert.strictEqual($items.eq(1).hasClass(LIST_ITEM_SELECTED_CLASS), true, 'item has selected class');
+
+        assert.strictEqual(this.onSelectionChangedSpy.callCount, 2, 'OnSelectionChanged should be called on every click');
+    });
+
+    QUnit.test('onSelectionChanged event shouldn`t be called on item click if selectionMode=single, selectByClick=false', function(assert) {
+        const $list = this.createList('single', false);
+        const $items = $list.find(`.${LIST_ITEM_CLASS}`);
+
+        $items.eq(0).trigger('dxclick');
+        assert.strictEqual($items.eq(0).hasClass(LIST_ITEM_SELECTED_CLASS), false, 'item doesn`t have selected class');
+        assert.strictEqual($items.eq(1).hasClass(LIST_ITEM_SELECTED_CLASS), false, 'item doesn`t have selected class');
+
+        $items.eq(1).trigger('dxclick');
+        assert.strictEqual($items.eq(0).hasClass(LIST_ITEM_SELECTED_CLASS), false, 'item doesn`t have selected class');
+        assert.strictEqual($items.eq(1).hasClass(LIST_ITEM_SELECTED_CLASS), false, 'item doesn`t have selected class');
+
+        assert.strictEqual(this.onSelectionChangedSpy.callCount, 0, 'OnSelectionChanged should not be called');
+    });
+
+    QUnit.test('onSelectionChanged event shouldn`t be called on item click when selectionMode=single, change selectByClick to true', function(assert) {
```

Comment:
For me test name really differs from what u test
```suggestion
    QUnit.test('selection should be changed on item click after selectByClick is set to true (selectionMode=single)', function(assert) {
```

Let's update all tests u added

Reply:
Fixed